app/controllers/admin/decks_controller.rb

```diff
@@ -24,6 +24,8 @@ def create
 
   def edit
     @deck = Deck.find(params[:id])
+    @cards = Card.all
+    @memberships = @deck.card_memberships
   end
```

Comment:
This can actually become:

```ruby
def edit
  @deck = Deck.find(params[:id])
  @cards = @deck.cards
end
```

This will give us access to the variable `@cards`, which will contain a list of `@deck`'s cards, in `@deck`'s edit view.

Reply:
Adding cards to and removing cards from decks, however, will be a little trickier since we need to destroy the card membership record from the database and **not** the card record (since we wanna keep those lil guys around for other decks). That will end up looking something like this:

```ruby
def add_card
  @deck = Deck.find(params[:deck_id])
  @card = Card.find(params[:card_id])

  @deck.cards << @card
end

def remove_card
  @deck = Deck.find(params[:deck_id])
  @card = Card.find(params[:card_id])

  @deck.cards.delete(@card)
end
```

Here I added two new actions to the `admin/decks_controller`, which is probably the cleanest way to go in my opinion. This is not exactly intuitive, so I'm happy to discuss further.